pkg/kube/controllers/extendedstatefulset/reconciler.go

```diff
@@ -258,3 +283,61 @@ func (r *ReconcileExtendedStatefulSet) getActualStatefulSet(ctx context.Context,
 
 	return result, maxVersion, nil
 }
+
+// listStatefulSetVersions gets all StatefulSets' versions and ready status owned by the ExtendedStatefulSet
+func (r *ReconcileExtendedStatefulSet) listStatefulSetVersions(ctx context.Context, exStatefulSet *essv1a1.ExtendedStatefulSet) (map[int]bool, error) {
+	result := map[int]bool{}
+
+	statefulSets, err := r.listStatefulSets(ctx, exStatefulSet)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, statefulSet := range statefulSets {
+		strVersion, ok := statefulSet.Annotations[essv1a1.AnnotationVersion]
+		if !ok {
+			strVersion = "0"
+		}
+
+		version, err := strconv.Atoi(strVersion)
+		if err != nil {
+			return result, errors.Wrapf(err, "Version annotation is not an int: %s", strVersion)
+		}
+
+		ready, err := r.isStatefulSetReady(ctx, &statefulSet)
+		if err != nil {
+			return nil, err
+		}
+
+		result[version] = ready
+	}
+
+	return result, nil
+}
+
+// isStatefulSetReady returns true if one owned Pod is running
+func (r *ReconcileExtendedStatefulSet) isStatefulSetReady(ctx context.Context, statefulSet *v1beta1.StatefulSet) (bool, error) {
+	podList := &api.PodList{}
+	err := r.client.List(
```

Comment:
Is there a way to get only the pods we own?
Seems inefficient to get ALL pods in the namespace.

Reply:
Use label selector to filer owned pods. I thought we can modify CRD and apply this selector to ex StatefulSet with its owned statefulSet.